src/panels/lovelace/editor/view-editor/hui-edit-view.ts

```diff
@@ -216,7 +216,9 @@ export class HuiEditView extends LitElement {
 
     const viewConf: LovelaceViewConfig = {
       cards: this._cards,
-      badges: this._badges!.map((entityConf) => entityConf.entity),
+      badges: this._badges!.map((entityConf) =>
+        Object.keys(entityConf).length > 1 ? entityConf : entityConf.entity
```

Comment:
We can save all new ones/edits as objects, as long as we keep loading strings to stay backwards compatible.

Reply:
is one really preferable over the other?

Reply:
I saves us from doing this check, keeps things simpler